tripy/tripy/frontend/tensor.py

```diff
@@ -205,51 +191,47 @@ def eval(self) -> Array:
         assert len(data) == 1, "Expects only one output from mlir_tensorrt.compiler executor"
         data = data[0]
         # Data is present now. Assign the underlying device type.
-        self.device = data.device
+        self.device = flat_ir.outputs[0].device
```

Comment:
can we instead query the `memref` device attribute to get the data type?

Reply:
Yes we could, but the code looks verbose:
```
device = "gpu" if memref.address_space == runtime.PointerType.device else "cpu"
```

Getting device from flat_ir is safe because it is the only source of truth, `executor` will also query from it and create memrefs respectively.